fov_processing_pipeline/wrappers.py

```diff
@@ -298,3 +299,96 @@ def qc_stats(df_stats, save_path):
     df_stats.to_pickle(save_path)
 
     return df_stats
+
+
+@task
+def data_splits(
+    df_stats,
+    save_dir,
+    split_names=["train", "validate", "test"],
+    split_amounts=[0.8, 0.1, 0.1],
+    group_column="ProteinDisplayName",
+    split_column="FOVId_rng",
+):
+    """
+    Given a stats dataframe, split each unique entry of `group_column` into train and test groups based on the `split_column random number, and save
+    those results in `save_dir` with the f"{save_dir}/{unique_group_column}_{train_or_test}.csv"
+
+    Parameters
+    ----------
+    df_stats: Dataframe
+        A stats dataframe with the columns corresponding to `group_column` and `split_column`
+
+    save_dir: str
+        Save directory for splits files
+
+    split_names: list of strs
+        Names of the splits  
+
+    split_amounts: list of floats
+        Fraction of data split that corresponds to split names.
+        Must be same length as split names and sum to 1.
+
+    group_column: str
+        Column over which to create cohorts that we split up
+
+    split_column: str
+        Column with values that we split in. All values must be in the range of [0, 1). See: numpy.random.rand
+
+    Returns
+    -------
+    splits_dict
+        A dictionary corresponding to split groups, split names, and the save file
+    """
+
+    # all these assertions should probably be raise.ValueError
+
+    # check for correct input
+    assert np.sum(split_amounts) == 1
```

Comment:
I'm a little confused. It looks like a bunch of your testing is being done here (below) directly in the function in wrappers.py rather than with all the other tests. Am I misunderstanding or is there a reason that that's desirable?

Reply:
I guess its because you're checking valid input parameters, rather than proper code functioning...